### PR TITLE
CaseCycle: add controller, api, request specs, swagger stuff

### PR DESCRIPTION
**Pull request by weedySeaDragon**
# 💅🏼 What issue does this fix?

#301 
## 🍂 Before You Submit

* [x] Did you write tests?
* [x] Did you run `bundle exec rspec` from the root?
* [x] Did you run `bundle exec rails rswag` from the root?
* [x] Did you run `bundle exec rubocop` from the root?
* [x] Did you run `yarn lint` in `/client`?
* [x] Did you run `yarn test` in `/client`?
* [ ] Are your primary keys UUIDs on any new tables?

## 🛷 Deployment Considerations

* [ ] Data Migrations
* [ ] Schema Migrations
* [ ] Dependencies

### CODE REVIEW

File: app/controllers/api/v1/case_cycles_controller.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    @case_cycle.destroy
+  end
+
+  private
+
+  def set_case_cycle
+    @case_cycle = CaseCycle.find_by!(slug: params[:slug])
+  end
+
+  # rubocop:disable Metrics/MethodLength
```
Comment (katelovescode) [inline app/controllers/api/v1/case_cycles_controller.rb:50]: Rubocop makes me crazy with this.  It's not a long method, it's a long list of params, get it together, rubocop.

File: spec/support/api/schemas/case_cycle.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    "copay_cents": { "type": "integer", "minimum": 0 },
+    "copay_currency": { "$ref": "currency_or_null.json" },
+    "copay_frequency": {
+      "anyOf": [
+        { "$ref":"copay_frequencies.json" },
+        { "type": "null" }
+      ]
+    },
+    "created_at": { "type": "string", "format": "date-time" },
+    "effective_on": { "$ref": "currency_or_null.json" },
```
Comment (katelovescode) [inline spec/support/api/schemas/case_cycle.json:23]: Should this be currency?  Looks like it should be a date.

File: spec/support/api/schemas/case_cycle.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    "expires_on": { "$ref": "currency_or_null.json" },
```
Comment (katelovescode) [inline spec/support/api/schemas/case_cycle.json:24]: Ditto on a couple more of these fields, should be dates?

File: spec/support/api/schemas/case_statuses.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "copay_frequencies": {
```
Comment (katelovescode) [inline spec/support/api/schemas/case_statuses.json:2]:
```suggestion
  "case_statuses": {
```

File: spec/swagger_helper.rb [hunk 5]
```diff
@@ -126,6 +141,46 @@
...
+                anyOf: [
+                  { type: :string, example: '123-45' },
+                  { type: :null }
+                ]
+              },
+              copay_cents: { type: :integer, minimum: 0 },
+              copay_currency: { type: :string, example: 'USD' },
+              copay_frequency: {
+                anyOf: [
+                  { '$ref': '#/components/schemas/copay_frequency' },
```
Comment (katelovescode) [inline spec/swagger_helper.rb:157]:
```suggestion
                  { '$ref': '#/components/schemas/copay_frequencies' },
```

File: spec/swagger_helper.rb [hunk 6]
```diff
@@ -126,6 +141,46 @@
...
+                  { type: :null }
+                ]
+              },
+              notified_on: {
+                anyOf: [
+                  { type: :string, example: '2019-06-30' },
+                  { type: :null }
+                ]
+              },
+              status: { '$ref': '#/components/schemas/case_status' },
```
Comment (katelovescode) [inline spec/swagger_helper.rb:179]:
```suggestion
              status: { '$ref': '#/components/schemas/case_statuses' },
```

**Review by katelovescode — CHANGES_REQUESTED** [review]
Couple missed copy/paste things, I think

File: spec/support/api/schemas/case_cycle.json (hunk 2 above)
Comment (weedySeaDragon) [reply spec/support/api/schemas/case_cycle.json:23]: yup - definitely a _date_

File: spec/support/api/schemas/case_cycle.json (hunk 3 above)
Comment (weedySeaDragon) [reply spec/support/api/schemas/case_cycle.json:24]: yup. most definitely.

File: spec/swagger_helper.rb (hunk 5 above)
Comment (weedySeaDragon) [reply spec/swagger_helper.rb:157]: 👍  I need to change the name of in the schema, too.

File: spec/swagger_helper.rb (hunk 6 above)
Comment (weedySeaDragon) [reply spec/swagger_helper.rb:179]: 👍   I need to update the name in the schema, too.
Good catch.

**Review by katelovescode — APPROVED** [review]

**Review by katelovescode — APPROVED** [review]

**Pull request merged**